### PR TITLE
NXP-27978: rework DocumentModelJsonReader

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

**Comment by troger** [comment]
T&P: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-troger/393/

File: nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java [hunk 1]
```diff
@@ -67,17 +67,54 @@
...
+        anySchema = true;
+    }
+
+    /**
+     * @deprecated since 11.1. Use {@link #ofSchemas(List)} instead.
+     */
+    @Deprecated(since = "11.1")
     public SimpleDocumentModel(List<String> schemas) {
-        this.schemas = new HashSet<>();
         anySchema = false;
```
Comment (RSalem07) [inline nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java]: is there a raison to initialise the `anySchema`, as it is a primitif boolean which is false by default

File: nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java [hunk 2]
```diff
@@ -67,17 +67,54 @@
...
+    public static SimpleDocumentModel ofSchemas(List<String> schemas) {
+        return new SimpleDocumentModel(schemas);
+    }
+
+    /**
+     * @deprecated since 11.1. Use {@link #ofSchemas(String...)} instead.
+     */
+    @Deprecated(since = "11.1")
+    public SimpleDocumentModel(String... schemas) {
+        this(Arrays.asList(schemas));
```
Comment (RSalem07) [inline nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java]: can be replaced by `List.of`

File: nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java [hunk 3]
```diff
@@ -67,17 +67,54 @@
...
+    }
+
+    /**
+     * @since 11.1
+     */
+    public static SimpleDocumentModel ofSchemas(String... schemas) {
+        return ofSchemas(Arrays.asList(schemas));
```
Comment (RSalem07) [inline nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java]: same as above

File: nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java (hunk 1 above)
Comment (troger) [reply nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java]: It's `final` so it needs initialization in the constructors.

Comment (RSalem07) [reply nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java]: yep, i didn't pay attention of the `final` keyword

**Comment by troger** [comment]
T&P: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-troger/394/

**Comment by troger** [comment]
T&P: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-troger/395/

**Pull request merged**